code/controllers/subsystem/atoms.dm

```diff
@@ -153,6 +156,15 @@ SUBSYSTEM_DEF(atoms)
 
 	return qdeleted || QDELING(A)
 
+/// Force reset atoms loc, as map expansion can botch turf contents for multitiles
+/datum/controller/subsystem/atoms/proc/fix_atoms_locs(list/atoms)
+	if(atoms)
+		for(var/atom/movable/A in atoms)
+			A.loc = A.loc
+	else
+		for(var/atom/movable/A in world)
+			A.loc = A.loc
+
```

Comment:
This isn't a good solution. Iterating over lists this large will likely just cause byond to put this on a sleeping loop, which if the list is large enough will lead to lag. Not good.

You'd need to talk to @Rohesie to know how to fix this properly, though my gut is to have a case for it for multitile objects in movable Initialize

Reply:
This is indeed horrible, it's just the best i could come up with without getting things ridiculously convoluted

Reply:
You can't handle this in Initialize, because it's already too late by then

It's not only about fixing the contents for the atom itself, but also making it available to others (airlocks and machinery linking anyone?). They might need it in their own Initialize, and there is no guarantee of ordering

Reply:
If it's a byond issue shouldn't we be bullying lummox about it?

Reply:
One solution to dodge all these problems and many more I could see would be to just run maploading in reverse, but I imagine if that's not been done there's a reason

Eg. Currently `_load_impl` iterates by increasing X, then decreasing Y. Iterating by decreasing X instead would ensure we expand map to its final size right away then just fill out the gaps

My only guess would be this isn't done out of performance considerations and smoother loading - performing big expansions in one go seems to actually be expensive for some reason

Reply:
> If it's a byond issue shouldn't we be bullying lummox about it?

well, that might work aswell !

Reply:
I'd be inclined to argue this is an engine oversight indeed given `loc = loc` fixes it